aptos-move/aptos-vm/src/delta_ext/delta_change_set.rs

```diff
@@ -184,6 +230,15 @@ impl ::std::iter::IntoIterator for DeltaChangeSet {
     }
 }
 
+impl<'a> ::std::iter::IntoIterator for &'a DeltaChangeSet {
```

Comment:
Honestly, instead of implementing an into iterator, I'd implement a take() on DeltaChangeSet and then chain the output with the appropriate iterator. A lot less lines and more verbose imho.
Also, what's up with the lifetimes?

Reply:
Re lifetimes - you need it since we have an iterator over references and compiler complains they may not live long enough.

Btw, I think this is not used atm, so can be dropped for now.